language/move-lang/tests/move_check/parser/spec_parsing_ok.move

```diff
@@ -0,0 +1,101 @@
+// These are some basic parsing tests for specifications which are expected to succeed.
+// Full testing of spec parsing correctness is done outside of this crate.
+//
+// Note that even though we only test parsing, we still need to ensure that the move code (not the specification)
+// is type checking correctly, because with no parsing error, the test harness
+// will run subsequent phases of the move-lang compiler.
+//
+// For parse failures, see the `spec_*_fail.move` test cases.
+
+module M {
+    synthetic expected_coin_sum: u64;
+    synthetic other: bool;
+
+    resource struct T{x: u64}
+    resource struct R{x: u64}
+
+    struct SomeCoin {
+        x : u64,
+        invariant x > 0,
+        y: u64,
+        invariant x == y,
+        invariant update old(y) < x,  // this weird coin only allows to increase it's value
+
+        // Tracking synthetic variable use case.
+        invariant update expected_coin_sum = expected_coin_sum - old(x) + x,
+        invariant pack expected_coin_sum = expected_coin_sum + x,
+        invariant unpack expected_coin_sum = expected_coin_sum - x,
+    }
+
+    fun with_aborts_if(x: u64)
+        aborts_if x == 0
+    {
+    }
+
+    fun with_ensures(x: u64): u64
+        ensures RET == x + 1
+    {
+        x + 1
+    }
+
+    fun with_multiple_conditions_and_acquires(addr: address, x: u64, y: u64): u64
+        acquires T, R
+        aborts_if y == 0
+        aborts_if 0 == y
+        ensures RET == x/y
```

Comment:
maybe just use the `return` keyword instead of RET?
It is an expression so it should work there.

But, @bob-wilson  would need to figure out the `return` with no arguments issue

Reply:
Right now this is just a naming convention. We can change this later once the `return` issue is solved.